src/components/pages/CompanyData/components/EditForm.tsx

```diff
@@ -0,0 +1,127 @@
+/********************************************************************************
+ * Copyright (c) 2021, 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+import {
+  Dialog,
+  DialogHeader,
+  DialogContent,
+  DialogActions,
+  Button,
+  CircleProgress,
+} from '@catena-x/portal-shared-components'
+import { Box } from '@mui/material'
+import { useState } from 'react'
+import { SiteForm } from './CreateSite/SiteForm'
+import { AddressForm } from './CreateAddress/AddressForm'
+import { useTranslation } from 'react-i18next'
+import {
+  type CompanyDataAddressType,
+  type CompanyDataSiteType,
+} from 'features/companyData/companyDataApiSlice'
+
+interface FormDetailsProps {
+  readonly open: boolean
+  readonly title: string
+  readonly description?: string
+  readonly handleClose: () => void
+  readonly isAddress?: boolean
+  readonly handleConfirm: () => void
+  readonly newForm?: boolean
+}
+
+export default function EditForm({
+  open,
+  title,
+  description,
+  handleClose,
+  isAddress = false,
+  handleConfirm,
+  newForm = false,
+}: FormDetailsProps) {
+  const { t } = useTranslation()
+  const [loading, setLoading] = useState<boolean>(false)
+  const [isValid, setValid] = useState<boolean>(false)
+  const handleSubmit = () => {
+    setLoading(false)
+    handleConfirm()
+  }
+  const handleSiteValidation = (
+    form: { body: CompanyDataSiteType } | undefined
+  ) => {
+    setValid(form ? true : false)
```

Comment:
Instead ternary operator write as expression that returns boolean:
`setValid(form !== undefined)`